torch/csrc/jit/passes/onnx/preclude_inlining.cpp

```diff
@@ -0,0 +1,87 @@
+#include <torch/csrc/jit/passes/onnx/preclude_inlining.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+
+namespace torch {
+namespace jit {
+
+void functionCallSubstitution(Block* block) {
+  for (auto it = block->nodes().begin(), end = block->nodes().end();
+       it != end;) {
+    Node* cur = *it++;
+    switch (cur->kind()) {
+      case prim::CallFunction: {
+        AT_ASSERT(cur->input(0)->node()->kind() == prim::Constant);
+        auto function_constant = cur->input(0)->node();
+        auto fun_type =
+            function_constant->output()->type()->expect<FunctionType>();
+
+        if ((fun_type->function()->qualname().qualifiedName().find(
+                 "torch.nn.functional") != std::string::npos) &&
+            (fun_type->function()->qualname().qualifiedName().find(
+                 "interpolate") != std::string::npos)) {
+          cur->removeInput(0);
+          Node* interpolate_node = block->owningGraph()->create(
+              Symbol::fromQualString("aten::__interpolate"),
+              {cur->inputs()},
+              cur->outputs().size());
+          interpolate_node->output()->copyMetadata(cur->output());
+          interpolate_node->insertAfter(cur);
+          cur->replaceAllUsesWith(interpolate_node);
+          cur->removeAllInputs();
+          cur->destroy();
+        } else {
+          cur->removeInput(0);
+          functionCallSubstitution(fun_type->function()->graph()->block());
+          GRAPH_UPDATE(
+              "Inlining in ONNX preclude inlining function '",
+              fun_type->function()->name(),
+              "' to ",
+              *cur);
+          GRAPH_UPDATE(
+              "Function in ONNX preclude inlining body: ",
+              *fun_type->function()->optimized_graph());
+          inlineCallTo(cur, fun_type->function(), false);
+        }
+      } break;
+      case prim::CallMethod: {
+        const std::string& name = cur->s(attr::name);
```

Comment:
Can you add a test for the new code path ?

Reply:
I'm adding a test in onnxruntime backend tests to cover this case (interpolate call within submodule). Let me know if you think of a more generalized way of testing this.